influxql/parser.go

```diff
@@ -2057,7 +2058,7 @@ func (p *Parser) peekRune() rune {
 	return r
 }
 
-func (p *Parser) parseSource() (Source, error) {
+func (p *Parser) parseSource(subqueries bool) (Source, error) {
```

Comment:
The reason I did this is because otherwise it would have required copying code. Splitting them into two different functions was impossible to compose because the subqueries logic had to go in the middle of the function. It wasn't possible to have it at the beginning or end.